src/core/main.c

```diff
@@ -461,6 +462,15 @@ static int parse_proc_cmdline_item(const char *key, const char *value, void *dat
                 if (arg_default_timeout_start_usec <= 0)
                         arg_default_timeout_start_usec = USEC_INFINITY;
 
+        } else if (proc_cmdline_key_streq(key, "systemd.watchdog_dev_path")) {
+
+                if (proc_cmdline_value_missing(key, value))
+                        return 0;
+
+                r = parse_path_argument_and_warn(value, false, &arg_watchdog_device_path);
+                if (r < 0)
+                        log_warning_errno(r, "Failed to parse watchdog device path: %s, ignoring.", value);
```

Comment:
parse_path_argument_and_warn() logs on its own (that's what the `_and_warn` suffix is supposed to say), no need to log here a second time, that'd just be confusing for the user